DrawerKit/DrawerKit/Internal API/PresentationController.swift

```diff
@@ -200,91 +318,63 @@ private extension PresentationController {
         animator.startAnimation()
     }
 
-    func addCornerRadiusAnimationEnding(at endPositionY: CGFloat, clamping: Bool = false) {
-        guard maximumCornerRadius > 0 && drawerPartialY > 0 && endPositionY != currentDrawerY else { return }
-
-        let endPosY = (clamping ? clamped(endPositionY) : endPositionY)
-        guard endPosY != currentDrawerY else { return }
+    func addCornerRadiusAnimationEnding(at endState: DrawerState) {
+        guard maximumCornerRadius != 0 && drawerPartialY != 0
+            && endState != currentDrawerState else { return }
 
         let animator = UIViewPropertyAnimator(duration: durationInSeconds,
                                               timingParameters: timingCurveProvider)
 
-        let endingCornerRadius = cornerRadius(at: endPosY)
+        let endingCornerRadius = cornerRadius(at: endState)
         animator.addAnimations {
             self.currentDrawerCornerRadius = endingCornerRadius
         }
 
-        if endPosY != drawerPartialY {
+        if endState == .collapsed || endState == .fullyExpanded {
             animator.addCompletion { _ in
                 self.currentDrawerCornerRadius = 0
             }
         }
 
         animator.startAnimation()
     }
-
-    func cornerRadius(at positionY: CGFloat) -> CGFloat {
-        guard maximumCornerRadius > 0 else { return currentDrawerCornerRadius }
-        guard drawerPartialY > 0 && drawerPartialY < containerViewH else { return 0 }
-        guard positionY >= 0 && positionY <= containerViewH else { return 0 }
-
-        let fraction: CGFloat
-        if supportsPartialExpansion {
-            if positionY < drawerPartialY {
-                fraction = positionY / drawerPartialY
-            } else {
-                fraction = 1 - (positionY - drawerPartialY) / (containerViewH - drawerPartialY)
-            }
-        } else {
-            fraction = 1 - positionY / containerViewH
-        }
-
-        return fraction * maximumCornerRadius
-    }
 }
 
 private extension PresentationController {
-    func endingPositionY(positionY: CGFloat, velocityY: CGFloat) -> CGFloat {
-        let isNotMoving = (velocityY == 0)
-        let isMovingUp = (velocityY < 0) // recall that Y-axis points down
-        let isMovingDown = (velocityY > 0)
-        let isMovingQuickly = (flickSpeedThreshold > 0) && (abs(velocityY) > flickSpeedThreshold)
+    func nextStateFrom(currentState: DrawerState, speedY: CGFloat) -> DrawerState {
+        let isNotMoving = (speedY == 0)
+        let isMovingUp = (speedY < 0) // recall that Y-axis points down
```

Comment:
What does `// recall that Y-axis points down` mean?

Reply:
It's somewhat counter-intuitive that a *negative* speed means moving *up* rather than down, and the reason is that the vertical axis is positive pointing down, hence the comment.